File: testdata/fizzbuzz.c
#include <stdio.h>
#include <stdlib.h>

char *fizzbuzz(int n) {
    if (n % 3 == 0 && n % 5 == 0) {
        return "FizzBuzz";
    } else if (n % 3 == 0) {
        return "Fizz";
    } else if (n % 5 == 0) {
        return "Buzz";
    }
    char *result = malloc(20 * sizeof(char));
    if (result == NULL) {
        perror("malloc");
        exit(EXIT_FAILURE);
    }
    sprintf(result, "%d", n);
    return result;
}

int parse_max(int argc, char *argv[]) {
    if (argc < 2) {
        return 100; // default max
    }
    return atoi(argv[1]);
}

int perform(int max) {
    for (int i = 1; i <= max; i++) {
        printf("%d: %s\n", i, fizzbuzz(i));
    }
    return 0;
}

int main(int argc, char *argv[]) {
    int max = parse_max(argc, argv);
    perform(max);
    return 0;
}

File: testdata/fizzbuzz.go
package main

import (
	"fmt"
	"os"
	"strconv"
)

func fizzbuzz(i int) string {
	switch {
	case i%15 == 0:
		return "FizzBuzz"
	case i%3 == 0:
		return "Fizz"
	case i%5 == 0:
		return "Buzz"
	default:
		return strconv.Itoa(i)
	}
}

func parseMax(args []string) int {
	if len(args) < 1 {
		return 100
	}
	max, err := strconv.Atoi(args[0])
	if err != nil || max < 1 {
		return 100
	}
	return max
}

func goMain(args []string) int {
	max := parseMax(args)
	for i := 1; i <= max; i++ {
		fmt.Printf("%d: %s\n", i, fizzbuzz(i))
	}
	return 0
}

func main() {
	status := goMain(os.Args[1:])
	os.Exit(status)
}
